include/envoy/router/router_ratelimit.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include "envoy/http/filter.h"
+#include "envoy/http/header_map.h"
+#include "envoy/ratelimit/ratelimit.h"
+
+namespace Router {
+/**
+* Base interface for reneric rate limit action.
+*/
+class Action {
+public:
+  virtual ~Action() {}
+
+  /**
+   * Potentially populate the descriptor array with new descriptors to query.
+   * @param route supplies the target route for the request.
+   * @param descriptors supplies the descriptor array to optionally fill.
+   * @param config supplies the filter configuration.
+   */
+  virtual void populateDescriptors(const RouteEntry& route,
```

Comment:
populateDescriptors is shared with the policy below so should be in a base class.